src/psyclone/domain/common/transformations/kernel_module_inline_trans.py

```diff
@@ -278,20 +293,21 @@ def _prepare_code_to_inline(
     @staticmethod
     def _get_psyir_to_inline(node):
         '''
-        Wrapper that gets the name and PSyIR of the routine or kernel
-        corresponding to the call described by `node`.
+        Wrapper that gets the PSyIR of the routine or kernel
+        corresponding to the call described by `node`. This supports calls to
+        routines or kernels which are polymorphic by returning a list of
+        Routine objects, as well as the associated interface symbol.
 
         :param node: the Call or CodedKern to resolve.
         :type node: :py:class:`psyclone.psyir.nodes.Call` |
                     :py:class:`psyclone.psyGen.CodedKern`
 
-        :returns: the name of the routine as seen by the caller and the
-                  PSyIR of the routine implementation.
-        :rtype: Tuple(str, :py:class:`psyclone.psyir.nodes.Call`)
+        :returns: the PSyIR of the routine implementation(s) and the associated
+            interface symbol if it is polymorphic.
+        :rtype: tuple[
+            list[:py:class:`psyclone.psyir.nodes.Routine`],
+            :py:class:`psyclone.psyir.symbols.GenericInterfaceSymbol`)]
```

Comment:
I would also return a union one or the other, not a tuple of one plus maybe the other, so basically:
```
if isinstance(node, CodedKern):
    return node.get_kernel_schedule()
...
```